Hadrons/Modules/MDistil/DistilMesonField.hpp

```diff
@@ -219,8 +236,49 @@ void TDistilMesonField<FImpl>::execute(void)
     std::map<std::string, DistilVector & > distVectors = {{"left",dvl}  ,{"right",dvr}};
     DistillationNoise &noisel = envGet( DistillationNoise , par().leftNoise);
     DistillationNoise &noiser = envGet( DistillationNoise , par().rightNoise);
-    std::map<std::string, DistillationNoise & >   noises = {{"left",noisel},{"right",noiser}};
-    
+
+    //auxiliar lambda functions
+    auto filenameDmfFn = [this, outpath](const unsigned int m, const unsigned int o, const int nl, const int nr)
+    {
+        std::stringstream ss;
+        ss << gamma_[o] << "_p";
+        for (unsigned int mu = 0; mu < momenta_[m].size(); ++mu)
+            ss << momenta_[m][mu] << ((mu == momenta_[m].size() - 1) ? "" : "_");
+
+        std::string filename = outpath + ss.str(); 
+        if(!isExact_)
+        {
+            filename += "_n" + std::to_string(nl) + "_" + std::to_string(nr) ;
+        }
+        filename += "." + std::to_string(vm().getTrajectory()) + ".h5";
+        return filename;
+    };
+    auto metadataDmfFn = [this, &nt, &nVec, &noisel, &noiser](const unsigned int m, const unsigned int o, const int nl, const int nr)
+    {
+        DistilMesonFieldMetadata<FImpl> md;
+        for (auto pmu: momenta_[m])
+        {
+            md.Momentum.push_back(pmu);
+        }
+        md.Operator = gamma_[o];
+        md.Nt               = nt;   
+        md.Nvec             = nVec;     //nvec is the same for both sides
+        md.NoisePair        = {nl,nr};
+        md.MesonFieldType   = dmf_type_.at("left") + "-" + dmf_type_.at("right");
+        if(isExact_)
+        {
+            md.NoiseHashesLeft     = {"0"}; // exact distil convention
+            md.NoiseHashesRight    = {"0"};
```

Comment:
this is the first places where it probably produces something that's wrong, i.e. it does not save the noise hash for the stochastic field if it is contracted with an exact one.

If you want to keep the cases separate, a solution would be to have two booleans, isExactLeft and isExactRight and then act accordingly

Reply:
And I would to thing of a way to identify that too... because now I just have : stoch x exact = noise pairs x no noise pairs